proof/refine/ARM/Finalise_R.thy

```diff
@@ -2902,13 +2902,21 @@ crunch invs[wp]: prepareThreadDelete "invs'"
 
 end
 
+lemma weak_sch_act_wf_updates[simp]:
+  "weak_sch_act_wf sa (s\<lparr>ksReprogramTimer := a\<rparr>) = weak_sch_act_wf sa s"
+  "weak_sch_act_wf sa (s\<lparr>ksReleaseQueue := b\<rparr>) = weak_sch_act_wf sa s"
+  by (auto simp: weak_sch_act_wf_def tcb_in_cur_domain'_def)
+
+crunches tcbReleaseRemove
+  for weak_sch_act_wf[wp]: "\<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s"
```

Comment:
would it be possible to combine some of these crunches together? I imagine they are placed as they are needed, so it might not be possible, but if we can have a bigger crunches block where possible, that would be easier to maintain.

Reply:
They are a little bit strategic, it's difficult to combine them together without things breaking.

Reply:
I see. Yes, that is very much likely. Thanks for having a try anyway.